### ISSUE TITLE
dzsave `--skip-blanks` does not work for transparent background

### ISSUE THREAD

**Issue by TimPushkin** [libvips/libvips#3765]
**Describe the bug**
When I use `dzsave` with `--skip-blanks` to split a PNG with transparency fully transparent tiles are not removed.

**To Reproduce**
1. Use this image (everything around the black square is transparent):
![Test](https://github.com/libvips/libvips/assets/71890197/99ba7859-ef75-4f23-931f-145b2f5510a7)
2. Call `dzsave "Test.png" "test" --layout google --tile-size 256 --depth one --overlap 0 --background 0 --skip-blanks 0 --suffix .png`.

**Expected behavior**
Only the tiles that contain parts of the black square are created (`1/1`, `1/2`, `2/1`, `2/2`).

**Actual behavior**
All tiles are created, even the ones that are completely transparent.

**Environment**
 - OS: Windows 11
 - Vips: 8.15

**Comment by jcupitt** [comment]
Hi @TimPushkin,

Your transparent pixels are actually [255, 255, 255, 0], not just 0, so your `background` value won't work. 

I see:

```
$ vips dzsave transparent-square.png x --depth one --overlap 0 --skip-blanks 0 --background "255 255 255 0" --suffix .png
$ ls x_files/0/
1_1.png  1_2.png  2_1.png  2_2.png
```

**Comment by TimPushkin** [comment]
@jcupitt Thanks, using `--background "255 255 255 0"` indeed works as expected.

This is actually how Photoshop creates transparent PNGs. Maybe it would be a good idea to treat all colors with 0 alpha as equal, not sure though.

**Comment by TimPushkin** [comment]
Here is another test image, the black square is now at the bottom, everything else is transparent. Again created via Photoshop, so the transparent pixels are probably `255 255 255 0` again (not sure how to check this yet).
![Test](https://github.com/libvips/libvips/assets/71890197/84efe15b-5e7d-418c-b3e8-04d5653a7632)

With it if I use `vips dzsave "Test.png" "test255" --layout google --tile-size 256 --depth onetile --overlap 0 --background "255 255 255 0" --skip-blanks 0 --suffix .png` I get zoom levels 0 and 2 as expected (only non-transparent tiles), but level 1 again contains fully transparent tiles.

And if I use `vips dzsave "Test.png" "test0" --layout google --tile-size 256 --depth onetile --overlap 0 --background 0 --skip-blanks 0 --suffix .png`, as before, levels 0 and 1 are all right (no transparent tiles), but level 2 now contains transparent tiles again.

**Comment by jcupitt** [comment]
You can extract the alpha and use 0 for all the RGBs like this:

```
$ vips extract_band transparent-square.png alpha.png 3
$ vipsheader transparent-square.png 
transparent-square.png: 1024x1024 uchar, 4 bands, srgb, pngload
$ vips black rgb.png 1024 1024 --bands 3
$ vips bandjoin "rgb.png alpha.png" transparent-square-zero.png
```

And see the pixel values like this:

```
$ vips getpoint transparent-square-zero.png 0 0 
0 0 0 0 
$ vips getpoint transparent-square.png 0 0 
255 255 255 0 
```

I don't know cmd.exe well enough, but in bash you can get pixel (0, 0) and use that value as the background like this:

```
$ vips dzsave transparent-square.png x --depth one --overlap 0 --skip-blanks 0 --background "$(vips getpoint transparent-square.png 0 0)" --suffix .png
```

**Comment by jcupitt** [comment]
> This is actually how Photoshop creates transparent PNGs.

That's interesting. I wonder if it comes down to how the image was made? Did you start with a white square, then make it transparent and paint an area black? Perhaps if you start with a black square you'd get 0 in the transparent parts.

**Comment by jcupitt** [comment]
... you could also search for the first transparent pixel and use that as the background value, but it's pretty annoying at the command-line. You'd be better off with something like pyvips.

**Comment by TimPushkin** [comment]
Thanks for the suggested extract_band-bandjoin approach, seems to work well for my usecase. I also found a Photoshop pluging that lets me save transparent pixels as (0, 0, 0, 0) ([SuperPNG](https://www.fnordware.com/superpng/) if anyone else will struggle for this).

> That's interesting. I wonder if it comes down to how the image was made? Did you start with a white square, then make it transparent and paint an area black? Perhaps if you start with a black square you'd get 0 in the transparent parts.

The square was created black initially. Actually, I stumbled upon this when dividing an image imported from PDF into Photoshop --- such images are also saved with (255, 255, 255, 0) transparent pixels.
